hoot-core/src/main/cpp/hoot/core/algorithms/DirectionFinder.cpp

```diff
@@ -52,11 +52,27 @@ namespace hoot
 
 bool DirectionFinder::isSimilarDirection(const ConstOsmMapPtr& map, ConstWayPtr w1, ConstWayPtr w2)
 {
+  LOG_VARD(w1->getNodeIds());
+  LOG_VARD(w2->getNodeIds());
+
+  // skip empty ways
+  if (w1->getNodeIds().size() == 0 || w2->getNodeIds().size() == 0)
+  {
+    return false;
+  }
+  // check for shared start/end node combos that show reversal; fix for #2888
+  else if ((w1->getNodeIds()[0] == w2->getNodeIds()[0]) ||
+           (w1->getNodeIds()[w1->getNodeIds().size() - 1] ==
+            w2->getNodeIds()[w2->getNodeIds().size() - 1]))
+  {
+    return false;
+  }
+
```

Comment:
These changes still need to be tested against regression, since they could be fairly far reaching.